Mathlib/Combinatorics/Enumerative/Bell.lean

```diff
@@ -168,4 +175,33 @@ theorem uniformBell_eq_div (m : ℕ) {n : ℕ} (hn : n ≠ 0) :
   · exact Nat.mul_pos (Nat.pow_pos (Nat.factorial_pos n)) m.factorial_pos
   · rw [← mul_assoc, ← uniformBell_mul_eq _ hn]
 
+/-- The `n`th standard Bell number,
+which counts the number of partitions of a set of cardinality `n`. -/
+def standardBell : ℕ → ℕ
+  | 0 => 1
+  | n + 1 => ∑ k ∈ Finset.range (n + 1), Nat.choose n k * standardBell (n - k)
+
+@[simp]
+theorem standardBell_zero : standardBell 0 = 1 := by
+  unfold standardBell
+  rfl
+
+@[simp]
+theorem standardBell_one : standardBell 1 = 1 := by
+  unfold standardBell
+  simp only [zero_add, Finset.range_one, zero_tsub, Finset.sum_singleton,
+    choose_self, one_mul, standardBell_zero]
+
+@[simp]
+theorem standardBell_two : standardBell 2 = 2 := by
+  unfold standardBell
+  rw [Finset.range_succ]
+  simp only [Finset.range_one, Finset.mem_singleton, one_ne_zero, not_false_eq_true,
+    Finset.sum_insert, choose_self, tsub_self, one_mul, Finset.sum_singleton,
+    choose_succ_self_right, zero_add, tsub_zero, standardBell_zero, standardBell_one]
+
+theorem standardBell_succ (n : ℕ) :
```

Comment:
You should prove this first, since then you can use it instead of `unfold` above.